pkg/metrics/metrics_test.go

```diff
@@ -738,3 +738,73 @@ func TestProcessStartTimeMetricExist(t *testing.T) {
 
 	t.Fatalf("Metrics does not contain %v. Scraped content: %v", processStartTimeMetric, metricsFamilies)
 }
+
+func TestRecordVolumeGroupMetrics(t *testing.T) {
+	mgr, srv := initMgr()
+	srvAddr := "http://" + srv.Addr + httpPattern
+	defer shutdown(srv)
+	// add an operation
+	opKey := OperationKey{
+		Name:       "op1",
+		ResourceID: types.UID("uid1"),
+	}
+	opVal := NewOperationValue("driver", DynamicSnapshotType)
+	mgr.OperationStart(opKey, opVal)
+	// should create a Success data point with latency ~ 1100ms
+	time.Sleep(1100 * time.Millisecond)
+	success := &fakeOpStatus{
+		statusCode: 0,
+	}
+	mgr.RecordVolumeGroupMetrics(opKey, success, "driver")
+
+	// add another operation metric
+	opKey.Name = "op2"
+	opKey.ResourceID = types.UID("uid2")
+	mgr.OperationStart(opKey, opVal)
+	// should create a Failure data point with latency ~ 100ms
+	time.Sleep(100 * time.Millisecond)
+	failure := &fakeOpStatus{
+		statusCode: 1,
+	}
+	mgr.RecordVolumeGroupMetrics(opKey, failure, "driver")
+
+	expected :=
+		`# HELP snapshot_controller_operation_total_seconds [ALPHA] Total number of seconds spent by the controller on an operation from end to end
+# TYPE snapshot_controller_operation_total_seconds histogram
+snapshot_controller_operation_total_seconds_bucket{driver_name="driver1",operation_name="op1",operation_status="Success",snapshot_type="",le="0.1"} 0
```

Comment:
For group snapshot, can you change the type from "snapshot_type" to "group_snapshot_type"?

Reply:
to create a new label, we have to replication the whole of operation metrics, which will unnecessarily replicate a lot, anyways, the value has to be same for it, what do you think?

Reply:
I added a comment to change to dynamic_group and pre-provisioned_group instead.
The individual volume snapshots are statically provisioned in this case, so I want to make the difference clear.